crates/types/src/config/common.rs

```diff
@@ -385,11 +436,33 @@ impl Default for CommonOptions {
                 Some(15),
                 Some(Duration::from_secs(5)),
             ),
-            auto_provision_partitions: true,
+            cluster_configuration: ClusterConfigurationType::default(),
+            default_cluster_configuration: None,
         }
     }
 }
 
+/// # Cluster Configuration
+#[serde_as]
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+#[serde(rename_all = "kebab-case")]
+pub enum ClusterConfigurationType {
+    #[default]
+    /// Use default cluster configuration
+    ///
+    /// This uses both `bootstrap-num-partitions` a
+    /// and `bifrost/default-provider` to seed the cluster
```

Comment:
This link specification looks a bit freestyle. Not sure whether users will be able to understand it. Maybe it is better to give a toml example

```
[bifrost]
default-provider = ...
```

Same with `bootstrap-num-partitions`.